### PR TITLE
UofT DSI | Applying Statistical Concepts - Assignment 1 (KNN Classification - wine class categories))

### PR DESCRIPTION
**Pull request by lee-data**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
My Assignment 1 submission for the module Applying Statistical Concepts. 

## What did you learn from the changes you have made?
I practiced KNN classification on a given wine dataset. 

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Yes - plenty. And I have questions as I compare code cell by cell from the assignment and the LIVE CODE notebooks, as well as my previous external Python class. 

1) In the lecture slides and LIVE CODE notebooks, the np.random.seed() was the given code. However, in the assignment 1, random.seed() was the given code. Since we imported given datasets (and not generating random numbers with either Numpy library nor Python's built-in function), does it matter which one to use? 

2) In "Classification-2.ipynb" LIVE CODE notebook, the given code to split the training set vs the test set is:
      "cancer_train, cancer_test = train_test_split(cancer, train_size=0.75, stratify=cancer["diagnosis"]"
 --> which uses panda's sklearn - a more robust and easier way to split data, and it stratifies data - which should be a better choice; 
 
However, in the assignment, the given code is:
 "split = np.random.choice([True, False], size=len(predictors_standardized), replace=True, p=[0.75, 0.25])" 
--> which is just random splitting, it does not stratifies, and therefore can lead to imbalanced splits. So why is this the given code in our assignment? 

3) The 'crosstab' confusion matrix is not asked in the assignment. Should we nevertheless use this in real life cases to build a good habit ? 

4) Should we also look at all "accuracy", "precision", and "recall" for every model, to build a good habit? Although I understand, as for the 'wine' classification, there probably isn't any significant consequence to false positive or false negative. We don't even know what these 0, 1, 2 are (e.g., are they wine types - like merlot / shiraz?; or are they bout how may years the wine has aged?) 

5) The LIVE CODE notebooks didn't standardize the predictor variables. I believe standardization is highly recommended best practice, so should we always do this even just for exercise to build a good habit? 

6) The assignment didn't ask to plot the output. Shouldn't we always plot it to build a good habit and find whether it's overfitting or underfitting? 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Not significantly, I referred to the codes given in LIVE CODE notebooks and my previous Python classes to practice coding. 

## How were these changes tested?
The codes were tested in Jupyter notebook. 

## A reference to a related issue in your repository (if applicable)
N/A

## Checklist
- [X] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by lee-data** [comment]
Closing this PULL REQUEST to update my assignment with a plot of accuracies grid.
